node/service_setup.go

```diff
@@ -60,6 +61,10 @@ func (node *Node) setupForBeaconLeader() {
 	node.serviceManager.RegisterService(service.BlockProposal, blockproposal.New(node.Consensus.ReadySignal, node.WaitForConsensusReady))
 	// Register client support service.
 	node.serviceManager.RegisterService(service.ClientSupport, clientsupport.New(node.blockchain.State, node.CallFaucetContract, node.getDeployedStakingContract, node.SelfPeer.IP, node.SelfPeer.Port))
+	// TODO(minhdoan): We will remove the old client support and use the new client support which uses new message protocol.
+	// Register client new support service.
+	node.serviceManager.RegisterService(service.RestClientSupport, restclientsupport.New(
```

Comment:
so, it only works for beacon leader?

Reply:
as of now - i can add todo later. 

the reason is that for demo we only deploy the lottery contract in beacon chain. in the future we should provide developers with capability to deploy contract in any shard and we should have client support for each shard.